### ISSUE TITLE
kinetics pretrained models

### ISSUE THREAD

**Issue by fmthoker** [BestJuly/Pretext-Contrastive-Learning#1]
Thanks for the code. Can you please share kinetics pretrained models. I want to include your paper in my experimental study.

**Comment by BestJuly** [comment]
Hi, @fmthoker. I have uploaded the requested checkpoint to [google drive](https://drive.google.com/file/d/1iVnAhVZOAR76PUXt0g4apNksWahB7nAT/view?usp=sharing), which is pretrained using our methods and can be used to downstream tasks by finetuning. Please note that the reported retreival accucacies in the paper are not based on this model, but you can still get good retrieval accuracies. For this part, you need to modify the `retrieve_clips.py`.

If you have other usage in your own experiments, also please pay attention to the checkpoint dict. You may need to write a function similar to [this](https://github.com/BestJuly/Pretext-Contrastive-Learning/blob/master/retrieve_clips.py#L29) to adjust names.

There are two different kinds of R2plus1D in my codes and R(2+1)D-18-3D is defined [here](https://github.com/BestJuly/Pretext-Contrastive-Learning/blob/master/models/network.py#L147).

The current `retrieve_clips.py` and `ft_classify.py` still do not support R(2+1)D-18-3D. But the usage is similar to other network backbones and it would be easy to modify for experiments.

If you need further help, I am glad to share part of my code here. However, the update of this repo might be delayed because recently, I am afraid I could not find time to refactor codes for a published version.

**Comment by fmthoker** [comment]
Thanks for sharing the models. I would try to adapt retrieve_clips.py and ft_classify.py for R(2+1)D-18-3D and get back to you in case of any problems. One question though, you mention that there are 2 R2plus1D models, is there a difference between the two?

**Comment by BestJuly** [comment]
Yes, the main differences lie to the model depth. R2plus1D means models with similar architecture to ResNet but using (2+1)D instead of 3D. R3D and R21D were mentioned in some previous works such as VCP/VCOP/etc., where the network depth is shallower compared to ResNet-18-3D (in some papers, they call 3D-ResNet-18). The layer numbers for each block of ResNet-18 is `(2,2,2,2)` while in R3D and R21D, these numbers are `(1,1,1,1)`. You can check the layer sizes [here](https://github.com/BestJuly/Pretext-Contrastive-Learning/blob/master/train_vcp_contrast.py#L276).

To use ResNet-18-3D and R2Plus1D, a convenient way is to directly use models provided by `torchvision` with `pretrained=False` for self-supervised learning. Therefore, the network archtecture of my provided model weights are defined in [network.py](https://github.com/BestJuly/Pretext-Contrastive-Learning/blob/master/models/network.py#L147), **NOT** in [r21d.py](https://github.com/BestJuly/Pretext-Contrastive-Learning/blob/master/models/r21d.py).
